arangosh/Utils/ManagedDirectory.cpp

```diff
@@ -637,4 +637,20 @@ ssize_t ManagedDirectory::File::offset() const {
   return fileBytesRead;
 }
 
+void ManagedDirectory::File::skip(size_t count) {
+  // TODO is there a better implementation than just read count bytes?
+  // how does this work with gzip?
+  size_t const bufferSize = 4 * 1024;
+  char buffer[bufferSize]; // alloc on the heap?
+
+  while (count > 0) {
+    TRI_read_return_t bytesRead = read(buffer, std::min(bufferSize, count));
+    if (bytesRead == 0) {
```

Comment:
On error, `read(...)` returns `-1` and sets `_status`. This case is not handled here, and if it happens, everything will :boom: